text/src/swam/witx/WitxParser.scala

```diff
@@ -0,0 +1,53 @@
+package swam
+package witx
+
+import java.nio.file.Path
+
+import fastparse._
+import swam.text.{ParserException, readFile}
+import swam.witx.parser.{ImportContext, ModuleParser, TypesParser}
+import swam.witx.unresolved.{BaseWitxType, ModuleInterface}
+import cats.effect._
+import cats.implicits._
+
+/**
+  * @author Javier Cabrera-Arteaga on 2020-03-20
+  */
+class WitxParser[F[_]](implicit val F: Effect[F]) {
+
+  def parseModuleInterface(file: Path, blocker: Blocker, ctx: ImportContext[F], chunkSize: Int = 1024)(
+      implicit cs: ContextShift[F]): F[(Map[String, BaseWitxType], ModuleInterface)] =
+    for {
+      input <- readFile(file, blocker, chunkSize)
+      interface <- parseModuleString(input, ctx)
+    } yield interface
+
+  private[swam] def parseModuleString(input: String,
+                                      ctx: ImportContext[F]): F[(Map[String, BaseWitxType], ModuleInterface)] =
+    F.liftIO {
```

Comment:
There is no need to go through an `IO`. It was an error in the previous parser, which is also fixed on master. You can do the same.